### ISSUE TITLE
Testcase "No response" when URL using HTTPS (TLS 1.0+)

### ISSUE THREAD

**Issue by diogouchoas** [Qarj/WebInject#38]
I'm having trouble making a testcase work using HTTPS.

```
<testcases repeat="1">
<case
    id="1"
    description1='try login on app'
    method="post"
    posttype="application/json"
    postbody='{"login": "exampleuser@gmail.com","password": "ElS086sN70I3961N9KmF"}'
    url="https://app.example.com/api/v2/auth/login"
/>
</testcases>
```

The output is:
```
Starting WebInject Engine...

-------------------------------------------------------
Test:  app_login.xml - 1 
try login on app 
 POSTURL /api/v2/auth/login 
Failed - No Response 
TEST CASE FAILED
Response Time = 0.01 sec 
------------------------------------------------------- 
Start Time: Wed 11 Oct 2017, 15:04:11
Total Run Time: 0.079 seconds

Total Response Time: 0.010 seconds

Test Cases Run: 1
Test Cases Passed: 0
Test Cases Failed: 1
Verifications Passed: 0
Verifications Failed: 1
```

When using http on the url it works, but using https I always get a "No Response".

Tried the same request using curl and it works using http and https.

**Comment by Qarj** [comment]
I'm presuming this is some type of internal application? Or do you have a publicly available website using the same SSL certificate?

Are you able to use WebInject to get this url `https://www.totaljobs.com` ?

**Comment by Qarj** [comment]
@diogouchoas Can you please do a git pull then try and run this new self test:

```
perl webinject.pl selftest/substeps/ssl.xml 
```

Both steps should pass, the second step is a JSON post over SSL and works for me.

Can you please also do a `perl -version` and let me know the first full line of output.

I've just tried this self test on Ubuntu 16.04 LTS and it didn't work for me, so I'm going to take a look and try and figure out what is going on. It works fine on Windows using Strawberry Perl.

**Comment by Qarj** [comment]
Inexplicably the `LWP::Protocol:https` library does not come pre installed on Ubuntu 16.04 LTS and most probably other flavours of Linux. I've updated the instructions for Linux to include:

```
sudo cpan LWP::Protocol:https
```

I'm going to test a few other distros to check that they work with SSL (after following the install instructions).

**Comment by Qarj** [comment]
Ok I've now tested the new instructions with Fedora 26, Ubuntu 16.04, Linux Mint 18 and 18.1, OS X El Capitan and macOS 10.12 and of course Windows with Strawberry Perl. All self tests are passing including the new SSL test.

To get this working with Mac and perlbrew, I had to make a few tweaks to webinject.pl (another existing issue), but really the main problem is that some Linux distros do not include `LWP::Protocol::https`.
